instrumentation/apache-camel-2.20/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/apachecamel/aws/SqsCamelTest.groovy

```diff
@@ -5,544 +5,135 @@
-            "messaging.destination" "sqsCamelTest"
-          }
-        }
-        span(2) {
-          name "SQS.SendMessage"
-          kind PRODUCER
-          childOf span(1)
-          attributes {
-            "aws.agent" "java-aws-sdk"
-            "aws.endpoint" "http://localhost:$sqsPort"
-            "aws.operation" "SendMessage"
-            "aws.queue.url" "http://localhost:$sqsPort/queue/sqsCamelTest"
-            "aws.service" "AmazonSQS"
-            "http.flavor" "1.1"
-            "http.method" "POST"
-            "http.status_code" 200
-            "http.url" "http://localhost:$sqsPort"
-            "net.peer.name" "localhost"
-            "net.peer.port" sqsPort
-            "net.transport" "IP.TCP"
-          }
-        }
-        span(3) {
-          name "SQS.ReceiveMessage"
-          kind CONSUMER
-          childOf span(2)
-          attributes {
-            "aws.agent" "java-aws-sdk"
-            "aws.endpoint" "http://localhost:$sqsPort"
-            "aws.operation" "ReceiveMessage"
-            "aws.queue.url" "http://localhost:$sqsPort/queue/sqsCamelTest"
-            "aws.service" "AmazonSQS"
-            "http.flavor" "1.1"
-            "http.method" "POST"
-            "http.status_code" 200
-            "http.url" "http://localhost:$sqsPort"
-            "http.user_agent" String
-            "net.peer.name" "localhost"
-            "net.peer.port" sqsPort
-            "net.transport" "IP.TCP"
-          }
-        }
-        span(4) {
-          name "sqsCamelTest"
-          kind INTERNAL
-          childOf span(2)
-          attributes {
-            "apache-camel.uri" "aws-sqs://sqsCamelTest?amazonSQSClient=%23sqsClient"
-            "messaging.destination" "sqsCamelTest"
-            "messaging.message_id" String
-          }
-        }
+    assertTraces(4) {
+      trace(0, 1) {
+        AwsSpan.sqs(it, 0, "SQS.ListQueues")
       }
-      trace(1, 1) {
-        span(0) {
-          name "SQS.ReceiveMessage"
-          kind CLIENT
-          hasNoParent()
-          attributes {
-            "aws.agent" "java-aws-sdk"
-            "aws.endpoint" "http://localhost:$sqsPort"
-            "aws.operation" "ReceiveMessage"
-            "aws.queue.url" "http://localhost:$sqsPort/queue/sqsCamelTest"
-            "aws.service" "AmazonSQS"
-            "http.flavor" "1.1"
-            "http.method" "POST"
-            "http.status_code" 200
-            "http.url" "http://localhost:$sqsPort"
-            "net.peer.name" "localhost"
-            "net.peer.port" sqsPort
-            "net.transport" "IP.TCP"
-          }
-        }
+      trace(1, 5) {
+        CamelSpan.direct(it, 0, "input")
+        CamelSpan.sqsProduce(it, 1, queueName, span(0))
+        AwsSpan.sqs(it, 2, "SQS.SendMessage", queueUrl, null, PRODUCER, span(1))
+        AwsSpan.sqs(it, 3, "SQS.ReceiveMessage", queueUrl, null, CONSUMER, span(2))
+        CamelSpan.sqsConsume(it, 4, queueName, span(2))
```

Comment:
Is this span hierarchy correct? I thought that

```
Camel PRODUCER -> AWS PRODUCER -> AWS CONSUMER -> Camel CONSUMER
```

was the expected outcome, and here we have something like this:

```
Camel PRODUCER -> AWS PRODUCER --> AWS CONSUMER
                               \-> Camel CONSUMER
```

Is this something that was discussed earlier but I forgot about it? 😅

Reply:
🤷‍♂️